physics/GFS_rrtmgp_post.F90

```diff
@@ -0,0 +1,394 @@
+         fluxswDOWN_clrsky    ! RRTMGP shortwave clear-sky flux   (W/m2)
+    real(kind_phys), intent(in) :: &
+         raddt                ! Radiation time step
+    real(kind_phys), dimension(:,:), intent(in) :: &
+         aerodp,            & ! Vertical integrated optical depth for various aerosol species
+         cldsa,             & ! Fraction of clouds for low, middle, high, total and BL
+         cld_frac,          & ! Total cloud fraction in each layer
+         cldtaulw,          & ! approx 10.mu band layer cloud optical depth
+         cldtausw             ! approx .55mu band layer cloud optical depth
+    type(cmpfsw_type), dimension(:), intent(in) :: &
+         scmpsw               ! 2D surface fluxes, components:
+                              ! uvbfc - total sky downward uv-b flux at  (W/m2)
+                              ! uvbf0 - clear sky downward uv-b flux at  (W/m2)
+                              ! nirbm - downward nir direct beam flux    (W/m2)
+                              ! nirdf - downward nir diffused flux       (W/m2)
+                              ! visbm - downward uv+vis direct beam flux (W/m2)
+                              ! visdf - downward uv+vis diffused flux    (W/m2)
+
+
+    real(kind=kind_phys), dimension(:,:), intent(inout) :: fluxr
+
+    ! Outputs (mandatory)
+    real(kind_phys), dimension(:), intent(inout) :: &
+         tsflw,             & ! LW sfc air temp during calculation (K)
+         sfcdlw,            & ! LW sfc all-sky     downward flux   (W/m2)
+         sfculw,            & ! LW sfc all-sky     upward   flux   (W/m2)
+         nirbmdi,           & ! SW sfc nir    beam downward flux   (W/m2)
+         nirdfdi,           & ! SW sfc nir    diff downward flux   (W/m2)
+         visbmdi,           & ! SW sfc uv+vis beam downward flux   (W/m2)
+         visdfdi,           & ! SW sfc uv+vis diff downward flux   (W/m2)
+         nirbmui,           & ! SW sfc nir    beam upward   flux   (W/m2)
+         nirdfui,           & ! SW sfc nir    diff upward   flux   (W/m2)
+         visbmui,           & ! SW sfc uv+vis beam upward   flux   (W/m2)
+         visdfui,           & ! SW sfc uv+vis diff upward   flux   (W/m2)
+         sfcnsw,            & ! SW sfc all-sky     net      flux   (W/m2) flux into ground
+         sfcdsw               ! SW sfc all-sky     downward flux   (W/m2)
+    real(kind_phys), dimension(:,:), intent(inout) :: &
+         htrlw,             & ! LW all-sky heating rate (K/s)
+         htrsw,             & ! SW all-sky heating rate (K/s)
+         htrlwu               ! LW all-sky heating-rate updated in-between radiation calls.
+    type(sfcflw_type), dimension(:), intent(inout) :: &
+         sfcflw               ! LW radiation fluxes at sfc
+    type(sfcfsw_type), dimension(:), intent(inout) :: &
+         sfcfsw               ! SW radiation fluxes at sfc
+    type(topfsw_type), dimension(:), intent(out) :: &
```

Comment:
I don't know if this will lead to problems or not, but topfsw and topflw are not set to anything if doLWrad and doSWrad are F since they are intent(out).

Reply:
They aren't used anywhere not guarded by doLWrad and doSWrad, but you're correct. I cleaned this up